### PR TITLE
ARROW-15947: [R] rename_with s3 method for arrow_dplyr_query

### PR DESCRIPTION
**Pull request by markromanmiller**
Created a simple version of `rename_with` that applies the function to the current names of the .data argument and passes the result to `rename`.

This was a quick writeup on my end and passed a few smoke tests. I believe all the functions I've used in the new code are from base R or current imports like rlang, and the magic in editing the arrow data query is handled by `rename`.

Let me know what next steps can be, like writing test cases, etc.

### CODE REVIEW

**Comment by thisisnic** [comment]
Hi @MrMallIronmaker - thanks for submitting this PR!  Unit tests are definitely the way to go in terms of next steps.  Let us know if you have any questions about any of our set-up!

**Comment by MrMallIronmaker** [comment]
I've written a few test cases for `rename_with` and hit the "todo" for using tidyselect with select, as well. I think this is good enough to be reviewed! Thanks.

File: r/R/dplyr-select.R [hunk 1]
```diff
@@ -32,6 +32,16 @@ rename.arrow_dplyr_query <- function(.data, ...) {
 }
 rename.Dataset <- rename.ArrowTabular <- rename.RecordBatchReader <- rename.arrow_dplyr_query
 
+rename_with.arrow_dplyr_query <- function(.data, .fn, .cols = everything(), ...) {
+  .fn <- rlang::as_function(.fn)
```
Comment (thisisnic) [inline r/R/dplyr-select.R:36]: In `arrow/r/R/arrow-package-R` we have a load of `rlang` function listed with `@importFrom` roxygen tags.  Could you add `as_function` there and then call it directly here?

**Review by thisisnic — COMMENTED** [review]
Thanks for submitting this PR - it'll be great to have this functionality added!

A few minor comments and then also, please could you add an extra test where `.cols` doesn't end up selecting anything?

**Comment by MrMallIronmaker** [comment]
Changes made as requested - thanks!

**Review by thisisnic — COMMENTED** [review]
Great - thanks for making those changes!  One last tiny thing - the linter is complaining about spacing; would you mind running the styler on the files you've changes so that it automatically fixes those?  More information on how to do that can be found here in the docs: https://arrow.apache.org/docs/r/articles/developers/workflow.html

**Comment by markromanmiller** [comment]
I changed the spacing manually based on the lintr job output. `remotes::install_github` wasn't able to download the tarball of the lintr branch mentioned in the workflow setup, but that issue it seems to be a Github thing? I wasn't able to download it either through the web interface. In any case, I was able to run the standard lintr, and there were no linting messages in the files I've changed.

**Review by jonkeane — APPROVED** [review]
This is looking great. Thank you for your contribution!

**Pull request closed without merging**